### PR TITLE
gigatron: hook up screen, change machine type

### PR DESCRIPTION
**Pull request by Sterophonick**
thanks AJRhacker

### CODE REVIEW

**Comment by Robbbert** [comment]
COMP macro is incorrect - should be
COMP(2018, gigatron,    0,     0, gigatron, gigatron, gigatron_state, empty_init, "Marcel van Kervinck", "Gigatron TTL Microcomputer", MACHINE_IS_SKELETON) 

also, the file is missing its terminating newline - you can see the red mark there.

**Comment by Sterophonick** [comment]
Should be fixed.

**Comment by Sterophonick** [comment]
Oh, how do I provide you guys with the ROM from the Gigatron?

File: src/mame/drivers/gigatron.cpp [hunk 1]
```diff
@@ -50,6 +67,12 @@ void gigatron_state::gigatron(machine_config &config)
 	m_maincpu->set_addrmap(AS_PROGRAM, &gigatron_state::prog_map);
 	m_maincpu->set_addrmap(AS_DATA, &gigatron_state::data_map);
 
+	/* video hardware */
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
```
Comment (ajrhacker) [inline src/mame/drivers/gigatron.cpp]: This needs a definition for the `screen_update` method. A stub will do for now.

File: src/mame/drivers/gigatron.cpp [hunk 2]
```diff
@@ -24,8 +27,14 @@ class gigatron_state : public driver_device
 	}
 
 	void gigatron(machine_config &config);
+	
+	void init_gigatron();
```
Comment (ajrhacker) [inline src/mame/drivers/gigatron.cpp]: This is a prototype for an undefined function which isn't even necessary at the moment.

**Comment by Sterophonick** [comment]
Done.

File: src/devices/cpu/gigatron/gigatron.cpp [hunk 3]
```diff
@@ -11,7 +11,7 @@
 #include "gigatrondasm.h"
 
 
-DEFINE_DEVICE_TYPE(GTRON, gigatron_cpu_device, "gigatron", "GTRON")
+DEFINE_DEVICE_TYPE(GTRON, gigatron_cpu_device, "gigatron", "Gigatron CPU Device")
```
Comment (Tafoid) [inline src/devices/cpu/gigatron/gigatron.cpp]: Currently there is a machine shortname conflict here.  "gigatron" cannot be both the actual machine name and a device name as devices are handled as machines internally and in XML output.  Please rename it to something like "gigatron_cpu" so the duplication can be fixed.

Comment (Sterophonick) [reply src/devices/cpu/gigatron/gigatron.cpp]: Fixed.

**Pull request merged**